patches/tModLoader/Terraria/Recipe.cs.patch

```diff
@@ -153,6 +153,22 @@
  				if (num <= 0)
  					continue;
  
+@@ -138,6 +_,7 @@
+ 						break;
+ 
+ 					if (item.IsTheSameAs(item2) || useWood(item.type, item2.type) || useSand(item.type, item2.type) || useFragment(item.type, item2.type) || useIronBar(item.type, item2.type) || usePressurePlate(item.type, item2.type) || AcceptedByItemGroups(item.type, item2.type)) {
++						RecipeLoader.ConsumedItems.Add(item.Clone());
```

Comment:
the clone should contain the stack size of how many were actually consumed.

Reply:
Yup, that makes sense.  I missed that.

Reply:
Fixed.  I had to add a few extra lines to track the stack number, but I made it as clean as I could.